components/utils.py

```diff
@@ -337,17 +340,19 @@ def _generate_ssl_certificate(ips,
     return cert_path, key_path
 
 
-def generate_internal_ssl_cert(ip):
+def generate_internal_ssl_cert(ip, additional_ips=[], dns_names=[]):
```

Comment:
please note that using mutable default arguments for functions is bad style in python, because they are only created once, and each function call will get the same list object. You are appending to `dns_names`, so for the first call, this function would receive `dns_names=[]`, but for the second call, it would receive `dns_names=['localhost']` (and for the third, `dns_names=['localhost', 'localhost']` etc).
This is only called once so the bug isn't visible, but it's better to guard against that.
The common idiom is to do in the function header `dns_names=None` and then inside the function body `if dns_names is None: dns_names = []`

(btw, i thought our linter checked for that?)